delegated_translator.go

```diff
@@ -88,17 +89,11 @@ func (dt *delegatedTranslator) find(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Translate URL by mapping `/providers/{CID}` to `/cid/{CID}`.
-	cidUrlParam := strings.TrimPrefix(r.URL.Path, "/providers/")
-	// TODO: replace with URL.JoinPath once upgraded to go 1.19
-	findByCid := path.Join("/cid", cidUrlParam)
-	if err != nil {
-		http.Error(w, "", http.StatusBadRequest)
-		return
-	}
-	uri, err := url.ParseRequestURI(findByCid)
-	if err != nil {
-		http.Error(w, "", http.StatusInternalServerError)
-		return
+	var cidUrlParam string
+	if encrypted {
+		cidUrlParam = strings.TrimPrefix(r.URL.Path, "/encrypted/providers/")
+	} else {
+		cidUrlParam = strings.TrimPrefix(r.URL.Path, "/providers/")
 	}
```

Comment:
Would it be better to just do this?
```suggestion
	cidUrlParam := path.Base(r.URL.Path)
```

Reply:
I _think_ using `path.Base` would return results for `/encrypted/fish/in/da/sea/<multihash>` right?

Reply:
It does what you want: https://go.dev/play/p/wdbYmr6j-l-

Reply:
Oo nice 👍